core/src/main/java/tc/oc/pgm/filters/LocationQueryModifier.java

```diff
@@ -0,0 +1,97 @@
+package tc.oc.pgm.filters;
+
+import javax.annotation.Nullable;
+import org.bukkit.Location;
+import org.bukkit.event.Event;
+import org.bukkit.util.Vector;
+import tc.oc.pgm.api.filter.Filter;
+import tc.oc.pgm.api.filter.query.LocationQuery;
+import tc.oc.pgm.filters.query.BlockQuery;
+
+public class LocationQueryModifier extends QueryModifier<LocationQuery> {
+
+  private final String[] coords;
```

Comment:
Break this up into multiple classes, right now you are parsing this field during every query. Same with the relative vs. absolute coordinates. Do parsing at XML time, but it creates different classes with different functionality.